### DISCUSSION TITLE
Acorn - BBC Micro

### DISCUSSION THREAD

**Discussion by GingerNinjaLab** [cc65/cc65#2482] [Ideas]
Would love there to be support for the BBC Micro in the list of targets.
These were very capable machines for the time and the community is still going strong.
There are some forks for the BBC platform floating around on the Internet but look like they have all been abandoned.

**Comment by Marrin** [comment]
Then write the necessary code or find people to do it.  There's already a memory configuration called `bbc.cfg`, so it can be used to write assembler already. 🙂

**Comment by Fabrizio-Caruso** [comment]
@GingerNinjaLab the problem is that no one has shown up so far to implement it. 
A few years ago @polluks was interested but I am not aware of any progress.
I am using vbcc compiler in the meantime but I would prefer CC65.
As mentioned above there is already a cfg file that could be used as starting point.

**Comment by polluks** [comment]
@Fabrizio-Caruso 
Sorry, no progress so far. However I will rearrange my todo list.
Do you know this current thread? https://www.stardot.org.uk/forums/viewtopic.php?t=29181

@Marrin 
ld65 does not create inf files right now.

**Reply by GingerNinjaLab** [reply]
Oh my goodness! Thats amazing! Thanks for the reply I will check that out.

I downloaded the repo: `https://github.com/dominicbeesley/cc65` and after some compilation on Linux and hacking some .s files to with some .define entries that were missing I did managed to get some samples to compile. It was not able to run them however. I suspect it is an address  issue, causing the program to crash out when you attempt to load it on the emulator.

**Reply by markjfisher** [reply]
Hi! I'm the author of the stardot thread!

I created a repo that's up to date (as of a couple of weeks ago, it's just 19 commits behind at the moment) with the current cc65 head here which is based on Dominic's work, we probably did the same thing, and maybe you missed my repo link in the thread:
https://github.com/markjfisher/cc65/tree/bbc, dominic's original work doesn't readily compile, which is why I've tried to pull it together.

I'd love to see this moving too.

It works, but as per my comments here https://www.stardot.org.uk/forums/viewtopic.php?p=432742&sid=cc7a3cc28acbfbe7ed3f7649e72d3b2b#p432742 it's a bit off being something that can currently merge into cc65.
(The comments in that thread are based on feedback I asked @oliverschmidt for on it).

**Comment by Fabrizio-Caruso** [comment]
@polluks Thanks for the link! Maybe someone is going to do it after all these years.
